samples/apply-mosaic-rule-to-rasters/src/main/java/com/esri/arcgismaps/sample/applymosaicruletorasters/components/ApplyMosaicRuleToRastersViewModel.kt

```diff
@@ -0,0 +1,192 @@
+/* Copyright 2025 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgismaps.sample.applymosaicruletorasters.components
+
+import android.app.Application
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.viewModelScope
+import com.arcgismaps.geometry.Point
+import com.arcgismaps.mapping.ArcGISMap
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.Viewpoint
+import com.arcgismaps.mapping.layers.RasterLayer
+import com.arcgismaps.raster.ImageServiceRaster
+import com.arcgismaps.raster.MosaicMethod
+import com.arcgismaps.raster.MosaicOperation
+import com.arcgismaps.raster.MosaicRule
+import com.arcgismaps.toolkit.geoviewcompose.MapViewProxy
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialogViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+/**
+ * Enum representing the available preset mosaic rule types for the sample.
+ */
+enum class MosaicRuleType(val label: String) {
+    ObjectID("Object ID"),
+    NorthWest("North West"),
+    Center("Center"),
+    ByAttribute("By Attribute"),
+    LockRaster("Lock Raster")
+}
+
+/**
+ * ViewModel for the Apply Mosaic Rule to Rasters sample.
+ */
+class ApplyMosaicRuleToRastersViewModel(app: Application) : AndroidViewModel(app) {
+    /**
+     * The ArcGISMap used in the sample, with a topographic basemap.
+     */
+    var arcGISMap by mutableStateOf(
+        ArcGISMap(BasemapStyle.ArcGISTopographic)
+    )
+
+    /**
+     * The ImageServiceRaster from the online raster image service.
+     */
```

Comment:
We don't usually use function level comments for properties

// comments should be sufficient for properties